support/util/pdb.go

```diff
@@ -1,22 +1,12 @@
 package util
```

Comment:
Pulling [this comment][1] down into a thread, because I don't understand:

> This pdbs prevent CI from going flake because of autoscaling relocating pods. If we get rid of the PDBs we need to prevent the pods from being drained by autoscaling in CI. To that end we could e.g. use the "cluster-autoscaler.kubernetes.io/safe-to-evict": "false" annotation

and I want to unpack that, but I don't want to clutter up the main comment stream, and we can mark this thread resolved once discussion wraps up.

If these pods are `SingleReplica` instead of `HighlyAvailable`, isn't it expected that they go unavailable while being shifted around due to autoscaling or anything else that wants to shuffle workload Pods around for whatever reason?  `cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` seems like it would achieve CI-stability with similar drain-blocking costs to the current `minAvailable: 1` PodDisruptionBudgets.  I'd expect us to:

* Drop the PDBs for `SingleReplica` (like this pull is currently doing in 4a82de8c4c),
* Remove any orphan PDBs that older versions of the HostedControlPlane controller had created for those `SingleReplica` clusters (e.g. so updating your HostedCluster from 4.17.0 to 4.18.hasThisFix wouldn't leave orphan PDBs lying around), and
* Improve any CI that assumes `SingleReplica` services will be highly available, so it can handle some discruption gracefully, without failing.  Maybe that's what [this comment][2] is imagining?

[1]: https://github.com/openshift/hypershift/pull/4857#issuecomment-2399402981
[2]: https://github.com/openshift/hypershift/pull/4857#issuecomment-2402202576